src/main/java/de/tum/in/www1/artemis/service/iris/IrisConstants.java

```diff
@@ -112,4 +112,89 @@ private IrisConstants() {
+             {{#user~}}{{tests}}{{~/user}}
+             
+             {{#system~}}Your chat history with the instructor:{{~/system}}
+             {{#each messages}}
+                 {{#if (equal this.role "user")}}
+                     {{#user~}}{{this.content}}{{~/user}}
+                 {{else}}
+                     {{#assistant~}}{{this.content}}{{~/assistant}}
+                 {{/if}}
+             {{/each}}
+             
+             {{#system~}}
+                 Do you understand what the instructor wants well enough to make suggestions to improve the exercise?
+                 It is okay to make some assumptions.
+                 If you have enough information to work with, say "1". Otherwise, say "0".
+             {{~/system}}
+             {{#assistant~}}{{gen 'will_suggest_changes' max_tokens=1}}{{~/assistant}}
+             
+             {{#if (contains will_suggest_changes "0")}}
+                 {{#system~}}Respond to the instructor and ask a question to clarify their intent.{{~/system}}
+                 {{#assistant~}}{{gen 'response' temperature=0.7 max_tokens=200}}{{~/assistant}}
+                 {{set 'components' ""}}
+             {{else}}
+                 {{#system~}}Respond to the instructor like a helpful assistant would, summarizing your plans for the exercise.{{~/system}}
+                 {{#assistant~}}{{gen 'response' temperature=0.7 max_tokens=200}}{{~/assistant}}
+                 
+                 {{#system~}}
+                     List the components that you mentioned you would make changes to, in order of priority, separated by commas.
+                     The possible components are: "problem statement", "solution", "template", and "tests".
+                     Say nothing else.
+                 {{~/system}}
+                 {{#assistant~}}{{gen 'components' temperature=0.0 max_tokens=30}}{{~/assistant}}
+                 {{#if (contains components 'problem statement')}}
+                     {{#system~}}How will you adapt the problem statement?{{~/system}}
+                     {{#assistant~}}{{gen 'ps_plan' temperature=0.5 max_tokens=100}}{{~/assistant}}
+                 {{/if}}
+                 {{#if (contains components 'solution')}}
+                     {{#system~}}How will you adapt the solution?{{~/system}}
+                     {{#assistant~}}{{gen 'solution_plan' temperature=0.5 max_tokens=100}}{{~/assistant}}
+                 {{/if}}
+                 {{#if (contains components 'template')}}
+                     {{#system~}}How will you adapt the template?{{~/system}}
+                     {{#assistant~}}{{gen 'template_plan' temperature=0.5 max_tokens=100}}{{~/assistant}}
+                 {{/if}}
+                 {{#if (contains components 'tests')}}
+                     {{#system~}}How will you adapt the tests?{{~/system}}
+                     {{#assistant~}}{{gen 'tests_plan' temperature=0.5 max_tokens=100}}{{~/assistant}}
```

Comment:
General feedback: Pyris currently only recognizes responses named `content`, so you'd have to adapt this.

Reply:
We need multiple responses. This will have to be a change to Pyris